src/libertem/analysis/point.py

```diff
@@ -13,12 +13,15 @@ def get_results(self, job_results):
                            desc="intensity of the integration over the selected point"),
         ])
 
+    def get_use_sparse(self):
+        return True
+
     def get_mask_factories(self):
         cx = self.parameters['cx']
         cy = self.parameters['cy']
 
         def _point_inner():
-            a = np.zeros(self.dataset.shape[2:])
-            a[int(cy), int(cx)] = 1
-            return a.astype(self.dtype)
+            a = sp.csr_matrix(((1,), ((int(cy),), (int(cx),))),
```

Comment:
Does this need to be a tuple here? I try to avoid tuples of length 1 in more complicated expressions, as the syntax can become a bit ugly. Maybe:

```suggestion
            a = sp.csr_matrix(([1], [[int(cy)], [int(cx)]]),
```

If this is equivalent.

Reply:
Indeed better, coming with next commit.

Reply:
The sparse matrix constructor was a bit picky and only accepted lists for the innermost part. In any case, more readable now, thx! :-) 👍